conjure/servicewriter.go

```diff
@@ -134,13 +161,44 @@ func hasAuth(endpoints []spec.EndpointDefinition) (hasHeaderAuth, hasCookieAuth
 	return
 }
 
-func serviceInterfaceAST(serviceDefinition spec.ServiceDefinition, info types.PkgInfo, withAuth, isClient bool) (astgen.ASTDecl, StringSet, error) {
+// Return true if all endpoints that require authentication are of the same auth type (header or cookie) and at least
+// one endpoint has auth. The same auth type is required because a single token provider will likely not be useful for
+// both auth types.
+func canAddTokenInterface(endpoints []spec.EndpointDefinition) bool {
+	numHeader := 0
+	numCookie := 0
+	for _, endpointDefinition := range endpoints {
+		if endpointDefinition.Auth == nil {
+			continue
+		}
+		possibleHeaderAuth, err := visitors.GetPossibleHeaderAuth(*endpointDefinition.Auth)
```

Comment:
Keep related logic together for readability. Example:

```go
if possibleHeaderAuth, err := visitors.GetPossibleHeaderAuth(*endpointDefinition.Auth); err != nil {
   return false
} else if possibleHeaderAuth {
    hasHeaderAuthEndpoint = true
}
```

Reply:
👍